src/modules/assets/services/assets.service.ts

```diff
@@ -119,10 +119,16 @@ export class AssetsService {
 		}
 	}
 
-	public async upload(assetFiletype: AssetFileType, file: Express.Multer.File): Promise<string> {
+	public async upload(
+		assetFiletype: AssetFileType,
+		file: Express.Multer.File,
+		prefferedKey?: string
+	): Promise<string> {
 		const parsedFilename = path.parse(file.originalname);
-		const key = `${assetFiletype}/${_.kebabCase(parsedFilename.name)}-${uuidv4()}${
-			parsedFilename.ext
+		const key = `${assetFiletype}/${
+			prefferedKey
+				? _.kebabCase(prefferedKey)
```

Comment:
i would move the kebabCase function to where you call upload
otherwise you're modifying the prefferedKey evan though the consumer of the upload function might want an exact key that will not be modified.